### ISSUE TITLE
No support about @@unscopables inheritance

### ISSUE THREAD

**Issue by yuki3** [heycam/webidl#1221]
https://webidl.spec.whatwg.org/commit-snapshots/02e834239089b82f8383cd35a234257076a712af/#interface-prototype-object

> Let unscopableObject be OrdinaryObjectCreate\(null\)\.
> 
> For each exposed member member of interface that is declared with the \[Unscopable\] extended attribute:​
> 
> Let id be member’s identifier\.
> 
> Perform \! CreateDataProperty\(unscopableObject, id, true\)\.
> 
> Let desc be the PropertyDescriptor\{\[\[Value\]\]:​ unscopableObject, \[\[Writable\]\]:​ false, \[\[Enumerable\]\]:​ false, \[\[Configurable\]\]:​ true\}\.
> 
> Perform \! DefinePropertyOrThrow\(interfaceProtoObj, @​@​unscopables, desc\)\.

The spec creates an unscopables object with the interface members with \[Unscopable\], which ends up hiding the inherited interface's unscopables object. This looks a spec issue to me.

Example)
```
interface Base {
  [Unscopable] void foo();
};

interface Derived : Base {
  [Unscopable] void bar();
};
```
will result in:
```
Base.prototype[Symbol.unscopables]
// => {foo: true}
Derived.prototype[Symbol.unscopables]
// => {bar: true}  // No entry of {foo: true}.
```

**Comment by domenic** [comment]
I agree this is an issue.

As discussed with @yuki3 in person, there are two solutions: we could make `Derived.prototype[Symbol.unscopables]` be `{foo: true, bar: true}`, or we could make `Derived.prototype[Symbol.unscopables]` be `{bar: true, __proto__: Base.prototype[Symbol.unscopables]}`.

I suspect the former is easier to implement and a bit simpler. (And theoretically probably a bit faster for engines to optimize, but, if you're using `with` anyway, you're probably already on the super-slow path.)

**Comment by annevk** [comment]
What happens in engines today? E.g., with https://dom.spec.whatwg.org/#parentnode?

**Comment by annevk** [comment]
Well yes, but it's mixed into `Element`, which `HTMLAnchorElement` for instance inherits from. In WebKit `HTMLAnchorElement.prototype[Symbol.unscopables` returns an object with eight properties set to true. Which I guess hints at @domenic's first solution.

cc @shvaikalesh

**Comment by petervanderbeken** [comment]
I don't think we have a lot of other things to base this choice on. I personally find the second solution slightly more logical, since it somewhat follows the interface prototype chain, but I guess the first solution is a bit more dictionary-like. I can't say I feel strongly either way.

**Comment by domenic** [comment]
There are no existing examples of this problem, which is why it's slipped by us so far. This got filed when we were contemplating adding some unscopable properties to HTMLElement, which would shadow the ones on Element, and that caused some test failures.
